blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/LocatorTest.java

```diff
@@ -0,0 +1,441 @@
+
+        // expect
+        assertTrue(locator.isValidDBKey("abc", "b"));
+    }
+
+    @Test
+    public void isValidDBKeyDbKeyContainsDelimiterReturnsTrue() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // expect
+        assertTrue(locator.isValidDBKey("a.b.c", "."));
+    }
+
+    @Test
+    public void isValidDBKeyEmptyDbKeyReturnsFalse() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // expect
+        assertFalse(locator.isValidDBKey("", "."));
+    }
+
+    @Test
+    public void isValidDBKeyEmptyDelimiterReturnsTrue() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // expect
+        assertTrue(locator.isValidDBKey("a.b.c", ""));
+    }
+
+    @Test
+    public void isValidDBKeyMultiCharDelimiterReturnsTrue() {
+
+        // given
+        Locator locator = new Locator();  // the method is not static, so we have to create an instance
+
+        // expect
+        assertTrue(locator.isValidDBKey("a.b.c", ".b."));
+    }
+
+    @Test
+    public void hashCodeNullReturnsZero() {
+
+        // given
+        Locator locator = new Locator();
+
+        // expect
+        assertEquals(0, locator.hashCode());
+    }
+
+    @Test
+    public void hashCodeReturnsHashCodeOfStringRepresentation() {
+
+        // given
+        Locator locator = Locator.createLocatorFromDbKey(fullyQualifiedName);
+
+        // expect
+        assertEquals(fullyQualifiedName.hashCode(), locator.hashCode());
+    }
+
+    @Test
+    public void equalsNullReturnsFalse() {
+
+        // given
+        Locator locator = new Locator();
+
+        // expect
+        assertFalse(locator.equals((Object)null));
+    }
+
+    @Test
+    public void equalsNonLocatorReturnsFalse() {
+
+        // given
+        Locator locator = new Locator();
+
+        // expect
+        assertFalse(locator.equals(new Object()));
+    }
+
+    @Test
+    public void equalsDifferentStringReturnsFalse() {
+
+        // given
+        Locator locator = Locator.createLocatorFromDbKey("a.b.c");
+        Locator other = Locator.createLocatorFromDbKey("a.b.d");
+
+        // expect
+        assertFalse(locator.equals((Object)other));
+    }
+
+    @Test
+    public void equalsSameStringReturnsTrue() {
+
+        // given
+        Locator locator = Locator.createLocatorFromDbKey("a.b.c");
+        Locator other = Locator.createLocatorFromDbKey("a.b.c");
+
+        // expect
+        assertTrue(locator.equals((Object)other));
+    }
+
+    @Test
+    public void equalsBothNullReturnsTrue() {
+
+        // given
+        Locator locator = new Locator();
+        Locator other = new Locator();
+
+        // expect
+        assertTrue(locator.equals((Object)other));
+    }
+
+    @Test
+    public void equalsThisNullThrowsException() {
```

Comment:
i might have missed this.  Is this test named correctly?  It doesn't throw an exception.